controller/volume_controller.go

```diff
@@ -1358,24 +1307,32 @@ func getRebuildingReplicaCount(e *longhorn.Engine) int {
 func (vc *VolumeController) getReplenishReplicasCount(v *longhorn.Volume, rs map[string]*longhorn.Replica) int {
 	usableCount := 0
 	for _, r := range rs {
+		// The failed to schedule local replica shouldn't be counted
+		if r.Spec.HealthyAt == "" && r.Spec.FailedAt == "" && r.Spec.NodeID == "" &&
```

Comment:
Put `!isDataLocalityDisabled(v)` in the first condition so we know it's designed to deal with the volume with data locality.

Reply:
Thank you. Fixed